pkg/controller/lifecycle/actuator.go

```diff
@@ -106,20 +119,16 @@ func (a *actuator) createSeedResources(ctx context.Context, log logr.Logger, clu
 	return a.createManagedResource(ctx, namespace, constants.ManagedResourceNamesControllerSeed, "seed", renderer, constants.NetworkProblemDetectorControllerChartNameSeed, namespace, values, nil)
 }
 
-func (a *actuator) createShootResources(ctx context.Context, log logr.Logger, cluster *controller.Cluster, namespace string) error {
+func (a *actuator) createShootResources(ctx context.Context, log logr.Logger, cluster *controller.Cluster, namespace string, pspDisabled bool) error {
 	defaultPeriod := 10 * time.Second
-	pspEnabled := true
 	pingEnabled := false
 	if a.serviceConfig.NetworkProblemDetector != nil {
 		if a.serviceConfig.NetworkProblemDetector.DefaultPeriod != nil {
 			defaultPeriod = a.serviceConfig.NetworkProblemDetector.DefaultPeriod.Duration
 		}
-		if a.serviceConfig.NetworkProblemDetector.PSPDisabled != nil {
-			pspEnabled = !*a.serviceConfig.NetworkProblemDetector.PSPDisabled
-		}
 	}
 
-	shootResources, err := a.getShootAgentResources(defaultPeriod, pingEnabled, pspEnabled)
+	shootResources, err := a.getShootAgentResources(defaultPeriod, pingEnabled, !pspDisabled)
```

Comment:
Please restore the former behaviour if `pspDisabled == false`. In this case it should be possible to disable PSP by the service config, i.e. supposed change:

```go
	pspDisabledByConfig := false
	pingEnabled := false
	if a.serviceConfig.NetworkProblemDetector != nil {
		if a.serviceConfig.NetworkProblemDetector.DefaultPeriod != nil {
			defaultPeriod = a.serviceConfig.NetworkProblemDetector.DefaultPeriod.Duration
		}
		if a.serviceConfig.NetworkProblemDetector.PSPDisabled != nil {
			pspDisabledByConfig = *a.serviceConfig.NetworkProblemDetector.PSPDisabled
		}
	}

	shootResources, err := a.getShootAgentResources(defaultPeriod, pingEnabled, !pspDisabled && !pspDisabledByConfig)
```

Reply:
The former behaviour still stands. I just moved it out.
But if it's cleaner your way, then let's go for it.